tests/Aspire.Hosting.MongoDB.Tests/MongoDbFunctionalTests.cs

```diff
@@ -0,0 +1,340 @@
+            {
+                await app.StartAsync();
+                try
+                {
+                    var hb = Host.CreateApplicationBuilder();
+
+                    hb.Configuration[$"ConnectionStrings:{db2.Resource.Name}"] = await db2.Resource.ConnectionStringExpression.GetValueAsync(default);
+
+                    hb.AddMongoDBClient(db2.Resource.Name);
+
+                    using (var host = hb.Build())
+                    {
+                        await host.StartAsync();
+
+                        await pipeline.ExecuteAsync(async token =>
+                        {
+                            var mongoDatabase = host.Services.GetRequiredService<IMongoDatabase>();
+
+                            var collection = mongoDatabase.GetCollection<Movie>(CollectionName);
+
+                            var results = await collection.Find(new BsonDocument()).ToListAsync(token);
+
+                            Assert.Collection(results,
+                                            item => Assert.Contains("The Shawshank Redemption", item.Name),
+                                            item => Assert.Contains("The Godfather", item.Name),
+                                            item => Assert.Contains("The Dark Knight", item.Name),
+                                            item => Assert.Contains("Schindler's List", item.Name)
+                                            );
+                        }, cts.Token);
+                    }
+                }
+                finally
+                {
+                    // Stops the container, or the Volume/mount would still be in use
+                    await app.StopAsync();
+                }
+            }
+        }
+        finally
+        {
+            if (volumeName is not null)
+            {
+                DockerUtils.AttemptDeleteDockerVolume(volumeName);
+            }
+
+            if (bindMountPath is not null)
+            {
+                try
+                {
+                    Directory.Delete(bindMountPath, recursive: true);
+                }
+                catch
+                {
+                    // Don't fail test if we can't clean the temporary folder
+                }
+            }
+        }
+    }
+
+    [Fact]
+    [RequiresDocker]
+    public async Task VerifyWithInitBindMount()
+    {
+        // Creates a script that should be executed when the container is initialized.
+
+        var dbName = "testdb";
+
+        var cts = new CancellationTokenSource(TimeSpan.FromMinutes(6));
+        var pipeline = new ResiliencePipelineBuilder()
+            .AddRetry(new() { MaxRetryAttempts = 10, BackoffType = DelayBackoffType.Linear, Delay = TimeSpan.FromSeconds(2) })
+            .Build();
+
+        var bindMountPath = Directory.CreateTempSubdirectory().FullName;
+
+        if (!OperatingSystem.IsWindows())
+        {
+            System.Diagnostics.Process.Start("/usr/bin/env", $"sudo chmod -R a+rwx {bindMountPath}").WaitForExit();
```

Comment:
Why can't this use File.SetUnixFileMode?

Reply:
Maybe we shouldn't use `Directory.CreateTempSubdirectory()` in this case since we need other users to be able to see it? `Directory.CreateTempSubdirectory()` explicitly only gives the current user access to the directory, for security reasons. We might as well make a normal folder if we need other users to access it.